sprint_14

Ramilyushka opened the pull request.

Ramilyushka committed:
Ramilia

Ramilyushka committed:
1. Фильтрация по поиску 2. Заглушки если нет трекеров и если поиск пустой

Ramilyushka committed:
Добавлен функционал:
отметка о выполненной трекер (плюс и галочка)
количество дней

Ramilyushka committed:
conflict?

conflict?

Ramilyushka committed:
Merge branch 'main' into sprint_14

Ramilyushka committed:
верстка AddTrackerViewController

Ramilyushka committed:
Merge branch 'sprint_14' of https://github.com/Ramilyushka/Tracker into sprint_14

* 'sprint_14' of https://github.com/Ramilyushka/Tracker:
  1
  Пытаюсь исправить

Ramilyushka committed:
маленькая ошибка

Ramilyushka committed:
Верстка экрана Создание трекера

Ramilyushka committed:
Верстка и логика добавления расписания

Ramilyushka committed:
Сделано добавление нового трекера

Ramilyushka committed:
Добавлено ограничение на количество символов в названии трекера

So1ovyov commented on Tracker/Constants.swift:
```diff
@@ -2,28 +2,10 @@
 
 import Foundation
 
 let ypFontMedium = "YSDisplay-Medium"
```
Давай добавим шрифты в `enum` чтобы они располагались в структуре
```
enum FontsString {
   static let ypFontMedium: String = "YSDisplay-Medium" 
}
```

So1ovyov commented on Tracker/CreateTracker/Cell/ScheduleTableCell.swift:
```diff
@@ -0,0 +1,72 @@
+    
+    static let reuseIdentifier = "scheduleCell"
+    
+    var selectedDay: Bool = false
```
Давай сделаем свойство приватным, а для взаимодействия или изменения его значения добавим специальный узко направленный метод `updateDay(isSelected: Bool)`

So1ovyov commented on Tracker/CreateTracker/Cell/TrackerCollectionViewHeader.swift:
```diff
@@ -0,0 +1,38 @@
+
+import UIKit
+
+class TrackerCollectionViewHeader: UICollectionReusableView {
```
Потерялся модификатор `final`

So1ovyov commented on Tracker/CreateTracker/Cell/TrackerCollectionViewHeader.swift:
```diff
@@ -0,0 +1,38 @@
+        super.init(frame: frame)
+        
+        addSubview(titleLabel)
+        titleLabel.translatesAutoresizingMaskIntoConstraints = false
```
Дублируется строка, ты ее уже написала выше, а эту можно удалить

So1ovyov commented on Tracker/CreateTracker/Cell/TrackerTableCell.swift:
```diff
@@ -0,0 +1,107 @@
+    
+    static let reuseIdentifier = "habitCell"
+    private let titleLabel = UILabel()
+    let subTitleLabel = UILabel()
```
Давай элементы интерфейса инкапсулировать внутри ячейки, а для ее настройки и изменения создадим публичный метод

So1ovyov commented on Tracker/CreateTracker/ScheduleViewController.swift:
```diff
@@ -0,0 +1,154 @@
+    }
+}
+
+extension ScheduleViewController {
```
Очень удобно выносить приватные методы в отдельное расширение, круто !

So1ovyov commented on Tracker/Resources/UIColor+Extensions.swift:
```diff
@@ -10,9 +10,30 @@ import UIKit
+    static var ypGreen1: UIColor { UIColor(named: "YP Green") ?? UIColor.green }
     static var ypBlue1: UIColor { UIColor(named: "YP Blue") ?? UIColor.red }
+    
+    static var ypColorSelection1: UIColor { UIColor(named: "Color selection 1") ?? UIColor.red }
```
Давай более предметно неймить константы, ypColorOrange, ypColorBlue и т.д. числа в коде ни о чем не говорят

So1ovyov commented on Tracker/Trackers/TrackerModels.swift:
```diff
@@ -10,62 +10,83 @@ import UIKit
 //сущность для хранения информации про трекер (для «Привычки» или «Нерегулярного события»)
 struct Tracker {
```
Предлагаю декомпозировать файл, создать для 1-ой модели 1 файл и все эти файлы вынести в папку `Models` таким образом, будет одно место где хранятся все модельки и все это будет в отдельных небольших файлах (таким образом мы не раздуваем один большой файл на много строк)

So1ovyov commented on Tracker/Trackers/TrackersViewController.swift:
```diff
@@ -14,14 +14,15 @@ final class TrackersViewController: UIViewController {
     private var completedTrackers: [TrackerRecord] = []
     
-    private var selectedDate: Int?
+    private var selectedDate = Date()
```
В ТЗ требуется использовать `private var selectedDate = Date()` давай такого формата и придерживаться

So1ovyov commented on ViewController.swift:
```diff
@@ -2,7 +2,7 @@
 //  ViewController.swift
```
Кажется этот файл попал по ошибке и никак не используется в проекте, давай избавимся от него

Ramilyushka committed:
Исправлены замечания по ревью
Дату в datePicker нельзя сделать в формате dd.mm.yy к сожалению @So1ovyov, максимум можно сделать dd.mm.yyyy

Ramilyushka committed:
Маленькое исправление

Ramilyushka committed:
изменила название selectedDate на currentDate

Ramilyushka committed:
Убрала кнопку в storyboard
